### PR TITLE
feat(table): Add Card Cell and hideColumns property

### PR DESCRIPTION
**Pull request by diogomateus**
### What this PR does
This PR adds the hideColumns property that allows hiding a column by prop setting.
This PR adds the Card Cell variant of cell rendering.
This PR adds minor fixes to styles breaking on website.

### How to test?

_Please include additional context on how to test this PR._

### Checklist:

- [ ] I reviewed my own code
- [ ] The changes align with the designs I received  
  Or give a reason why this does not apply:
- [ ] I have attached screenshot(s), video(s) or gif(s) showing that the solution is working as expected  
  Or give a reason why this does not apply:
- [ ] I have updated the task(s) status on Linear
- [ ] All new media is optimized (images, gifs, videos)

### Browser support

My code works in the following browsers:

- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by kimkwanka — APPROVED** [review]

File: src/lib/components/table/types.ts [hunk 1]
```diff
@@ -2,12 +2,16 @@ import { ReactNode } from 'react';
 import { BaseCellProps } from './components/TableCell/BaseCell/BaseCell';
 import { CTACellProps } from './components/TableCell/CTACell/CTACell';
 import { ButtonCellProps } from './components/TableCell/ButtonCell/ButtonCell';
+import { CardCellProps } from './components/TableCell/CardCell/CardCell';
 
-type BaseCellData = BaseCellProps & { type?: undefined };
-type CTACellData = CTACellProps & { type: 'CTA' };
-type ButtonCellData = ButtonCellProps & { type: 'BUTTON' };
+type DefaultCellProps = { colSpan?: number };
```
Comment (kimkwanka) [inline src/lib/components/table/types.ts:7]: Why is this needed here though?

Comment (diogomateus) [reply src/lib/components/table/types.ts:7]: Missed this comment before merging. Just for consistency, we might want to add a single text on all columns or even a single boolean value.
